### PR TITLE
Rename scalar Minimum and Maximum Ops

### PR DESCRIPTION
**Pull request by michaelosthege**
I'm not so sure if this is a good idea.

Maybe the other scalar Ops like `Add` should also be renamed?

### CODE REVIEW

**Review by brandonwillard — CHANGES_REQUESTED** [review]
This is a tough one, because the `Op`s are already prefixed by `scalar` via their package name, so there's a bit of redundancy to this change.  

Perhaps that informative package prefix is lost when objects are imported, but that could be retained/regained with something like `from theano.scalar.basic import maximum as scalar_maximum, Maximum as ScalarMaximum`.

**Comment by michaelosthege** [comment]
Changing the imports is just code style. Renaming the Ops is a different level.
Do you want to rename the Ops, or just change the import style? (From #55 I understood it was about a real rename.)

Personally, I strongly dislike import aliasing, because it can be really confusing. (Yes, I don't even alias `pandas`, `numpy` or `pyplot`.)
In this PR it also made refactoring unnecessarily hard. In one instance `theano.scalar` was aliased as `ts` and in many others as `scal` 😬

My preferred code style would be `from theano import scalar` without aliasing anything.

**Comment by michaelosthege** [comment]
@brandonwillard (how) do we want to move forward with this?

File: theano/tensor/basic.py [hunk 1]
```diff
@@ -1054,10 +1054,11 @@ def _scal_elemwise_with_nfunc(nfunc, nin, nout):
 
     def construct(symbol):
         symbolname = symbol.__name__
-        if not hasattr(scal, symbolname):
-            # some scalar Op equivalents are prefixed with "scalar_"
-            symbolname = f"scalar_{symbolname}"
         inplace = symbolname.endswith("_inplace")
+        if symbolname in "minimum,maximum,minimum_inplace,maximum_inplace":
```
Comment (twiecki) [inline theano/tensor/basic.py:1058]: TIL

**Comment by brandonwillard** [comment]
OK, I've rebased and refactored some of the utility functions that were duplicated between `theano.tensor` and `theano.scalar` (e.g. the `_scal_elemwise` decorator).

**Pull request merged**